### PR TITLE
vignettes 

### LINKED ISSUE #73: Add vignettes

### PR DESCRIPTION
**Pull request by G-Kazantzidis**
closes #73

### CODE REVIEW

**Comment by danielinteractive** [comment]
@shawnafang can you please review this PR maybe as discussed in standup? thank you

File: vignettes/Intro_to_jmpost.Rmd [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+The association between the longitudinal data and the overall survival can be achieved in different ways.
+The predicted SLD value or its first derivative over time can be directly treated as the linking factor.
+Other factors such as the time to growth, the tumor shrinkage rate or the tumor growth rate can also be used to associate longitudinal outcomes with overall survival.
+
+Bayesian methodology provides an important tool for the analysis of similar data.
+Firstly, the use of historical data from previous trials is improving the performance of the models.
+Secondly, prior definition allows the adjustment of the models according to the specific factors of each analysis.
+
+Although these statistical methods provide significant advantages in decision making processes in cancer drug development, their implementation is relatively intricate and time consuming.
+Compared to other R-packages, `jmpost` provides the flexibility of non-linear modeling for the longitudinal part of the joint model.
```
Comment (danielinteractive) [inline vignettes/Intro_to_jmpost.Rmd:54]:
```suggestion
Compared to other R-packages, `jmpost` provides the flexibility of non-linear mixed effects modeling for the longitudinal part of the joint model. 
```

File: vignettes/Intro_to_jmpost.Rmd [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+# Minimal usage 
+
+```{r}
+my_long_mod <- LongModel(stan = StanModule(
+    functions = "exp_long_functions.stan",
+    data = "exp_long_data.stan",
+    priors = long_prior(),
+    inits = list(),
+    parameters = "exp_long_parameters.stan",
+    transformed_parameters = "exp_long_transformed_parametes.stan"
```
Comment (danielinteractive) [inline vignettes/Intro_to_jmpost.Rmd:66]: can we avoid this for the minimal use?

File: vignettes/Intro_to_jmpost.Rmd [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+))
+
+my_temp_os <- LogLogisticOs()
+
+my_link <- HazardLink(
+    parameters = "beta_ttg",
+    contribution = "* rep_matrix(ttg(psi_ks, psi_kg, psi_phi), rows(time))",
+    stan = StanModule()
```
Comment (danielinteractive) [inline vignettes/Intro_to_jmpost.Rmd:74]: same here?

File: vignettes/Intro_to_jmpost.Rmd [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+```
+
+
+Call of the object prints the different sections.
+
+```{r Long_model print}
+Long_mod
+```
+
+The overall survival object contails all ther information for the paramtetrisation of the second sub-model of the joint model.
```
Comment (danielinteractive) [inline vignettes/Intro_to_jmpost.Rmd:228]:
```suggestion
The overall survival object contails all ther information for the parametrisation of the second sub-model of the joint model. 
```

File: vignettes/Intro_to_jmpost.Rmd [hunk 5]
```diff
@@ -0,0 +1,419 @@
...
+
+The association between the longitudinal data and the overall survival can be achieved in different ways.
+The predicted SLD value or its first derivative over time can be directly treated as the linking factor.
+Other factors such as the time to growth, the tumor shrinkage rate or the tumor growth rate can also be used to associate longitudinal outcomes with overall survival.
+
+Bayesian methodology provides an important tool for the analysis of similar data.
+Firstly, the use of historical data from previous trials is improving the performance of the models.
+Secondly, prior definition allows the adjustment of the models according to the specific factors of each analysis.
+
+Although these statistical methods provide significant advantages in decision making processes in cancer drug development, their implementation is relatively intricate and time consuming.
```
Comment (danielinteractive) [inline vignettes/Intro_to_jmpost.Rmd:53]: cite 2-3 other R-packages, like `jmbayes` or there is an INLA package for joint modeling

File: vignettes/Intro_to_jmpost.Rmd [hunk 6]
```diff
@@ -0,0 +1,419 @@
...
+Compared to other R-packages, `jmpost` provides the flexibility of non-linear modeling for the longitudinal part of the joint model. 
+Non-linear parametrization respects the biological implications of clinical trials in oncology where the observed tumor response usually follows non-linear patterns.
+
+# Minimal usage 
+
```
Comment (danielinteractive) [inline vignettes/Intro_to_jmpost.Rmd:58]: add some text

File: vignettes/Intro_to_jmpost.Rmd [hunk 7]
```diff
@@ -0,0 +1,419 @@
...
+    contribution = "* rep_matrix(ttg(psi_ks, psi_kg, psi_phi), rows(time))",
+    stan = StanModule()
+)
+
+my_os <- parametrize(osmod = my_temp_os, link = my_link)
+JointModel(long = my_long_mod, os = my_os)
+```
+
+
+# Common usage
```
Comment (danielinteractive) [inline vignettes/Intro_to_jmpost.Rmd:82]:
```suggestion
# Customizing the analysis
```

File: vignettes/Intro_to_jmpost.Rmd [hunk 8]
```diff
@@ -0,0 +1,419 @@
...
+my_link <- HazardLink(
+    parameters = "beta_ttg",
+    contribution = "* rep_matrix(ttg(psi_ks, psi_kg, psi_phi), rows(time))",
+    stan = StanModule()
+)
+
+my_os <- parametrize(osmod = my_temp_os, link = my_link)
+JointModel(long = my_long_mod, os = my_os)
+```
+
```
Comment (danielinteractive) [inline vignettes/Intro_to_jmpost.Rmd:80]: add text here what is going to be added to the package, few sentences

**Review by danielinteractive — COMMENTED** [review]
thanks George, please see inside

File: vignettes/Intro_to_jmpost.Rmd [hunk 9]
```diff
@@ -50,11 +50,14 @@ Firstly, the use of historical data from previous trials is improving the perfor
...
 
 Although these statistical methods provide significant advantages in decision making processes in cancer drug development, their implementation is relatively intricate and time consuming.
-Compared to other R-packages, `jmpost` provides the flexibility of non-linear mixed effects modeling for the longitudinal part of the joint model. 
+Compared to other R-packages such as `joineR` and `jmbayes`, `jmpost` provides the flexibility of non-linear mixed effects modeling for the longitudinal part of the joint model. 
 Non-linear parametrization respects the biological implications of clinical trials in oncology where the observed tumor response usually follows non-linear patterns.
 
 # Minimal usage 
 
+The minimal usage of the `jmpost` package calls example stan code already provided in the repository of the package. 
+The following chank illustrates the absolute minimal usage of the package.
```
Comment (danielinteractive) [inline vignettes/Intro_to_jmpost.Rmd:59]:
```suggestion
The following chunk illustrates the absolute minimal usage of the package.
```

File: vignettes/Intro_to_jmpost.Rmd [hunk 10]
```diff
@@ -0,0 +1,413 @@
+---
+title: "Introduction to `jmpost`"
+subtitle: "`J`oint `M`odel for `P`osterior `O`verall `S`urvival `T`ranjectories"
```
Comment (danielinteractive) [inline vignettes/Intro_to_jmpost.Rmd:3]:
```suggestion
subtitle: "`J`oint `M`odels for `P`redicting `O`verall `S`urvival `T`rajectories"
```

File: vignettes/Intro_to_jmpost.Rmd [hunk 11]
```diff
@@ -0,0 +1,413 @@
...
+# Introduction
+
+In clinical development of new cancer medicines a key challenge is to decide on the next steps after having collected the initial data in Phase 1 studies. 
+In phase 1 studies patients are followed until the cancer is growing again (progression event). 
+Thus, reliable overall survival data are not available. 
+However, we can use additional information from previous clinical trials or real-world data to understand the time between progression and death.
+Specifically, we can correlate the tumor response data, that are longitudinal measurements, (i.e. the shrinkage and growth of tumor lesions over time) with the overall survival of the patients or their hazard ratios [@beyer2020multistate]. 
+Thereby we can predict the overall survival in our Phase 1 studies and therefore make better decisions [@kerioui2020bayesian].
+
+The sum of the longest diameter of the cancer (SLD) provides an effective and representative biomarker for longitudinal measurements in solid cancer.
```
Comment (danielinteractive) [inline vignettes/Intro_to_jmpost.Rmd:41]:
```suggestion
The sum of the longest diameters (SLD) of the cancer lesions provides an effective and representative biomarker for longitudinal measurements in solid cancer. 
```

File: vignettes/Intro_to_jmpost.Rmd [hunk 12]
```diff
@@ -0,0 +1,413 @@
...
+
+In clinical development of new cancer medicines a key challenge is to decide on the next steps after having collected the initial data in Phase 1 studies. 
+In phase 1 studies patients are followed until the cancer is growing again (progression event). 
+Thus, reliable overall survival data are not available. 
+However, we can use additional information from previous clinical trials or real-world data to understand the time between progression and death.
+Specifically, we can correlate the tumor response data, that are longitudinal measurements, (i.e. the shrinkage and growth of tumor lesions over time) with the overall survival of the patients or their hazard ratios [@beyer2020multistate]. 
+Thereby we can predict the overall survival in our Phase 1 studies and therefore make better decisions [@kerioui2020bayesian].
+
+The sum of the longest diameter of the cancer (SLD) provides an effective and representative biomarker for longitudinal measurements in solid cancer. 
+SLD is relatively easy to measure with great accuracy while also being a good representative of the development of cancer.
```
Comment (danielinteractive) [inline vignettes/Intro_to_jmpost.Rmd:42]:
```suggestion
SLD is relatively easy to measure with reasonable accuracy while also being a good representative of the development of cancer. 
```

File: vignettes/Intro_to_jmpost.Rmd [hunk 13]
```diff
@@ -0,0 +1,413 @@
...
+Thus, reliable overall survival data are not available. 
+However, we can use additional information from previous clinical trials or real-world data to understand the time between progression and death.
+Specifically, we can correlate the tumor response data, that are longitudinal measurements, (i.e. the shrinkage and growth of tumor lesions over time) with the overall survival of the patients or their hazard ratios [@beyer2020multistate]. 
+Thereby we can predict the overall survival in our Phase 1 studies and therefore make better decisions [@kerioui2020bayesian].
+
+The sum of the longest diameter of the cancer (SLD) provides an effective and representative biomarker for longitudinal measurements in solid cancer. 
+SLD is relatively easy to measure with great accuracy while also being a good representative of the development of cancer. 
+
+The association between the longitudinal data and the overall survival can be achieved in different ways.
+The predicted SLD value or its first derivative over time can be directly treated as the linking factor.
```
Comment (danielinteractive) [inline vignettes/Intro_to_jmpost.Rmd:45]:
```suggestion
The predicted SLD value or its first derivative over time can be directly treated as the linking factor in the hazard model for survival.
```

File: vignettes/Intro_to_jmpost.Rmd [hunk 14]
```diff
@@ -0,0 +1,413 @@
...
+Although these statistical methods provide significant advantages in decision making processes in cancer drug development, their implementation is relatively intricate and time consuming.
+Compared to other R-packages such as `joineR` and `jmbayes`, `jmpost` provides the flexibility of non-linear mixed effects modeling for the longitudinal part of the joint model. 
+Non-linear parametrization respects the biological implications of clinical trials in oncology where the observed tumor response usually follows non-linear patterns.
+
+# Minimal usage 
+
+The minimal usage of the `jmpost` package calls example stan code already provided in the repository of the package. 
+The following chunk illustrates the absolute minimal usage of the package.
+
+```{r, eval=FALSE}
```
Comment (danielinteractive) [inline vignettes/Intro_to_jmpost.Rmd:61]: maybe add 1-line comments above each command in this code example to further help the reader understand it?

File: vignettes/Intro_to_jmpost.Rmd [hunk 15]
```diff
@@ -0,0 +1,413 @@
...
+
+my_temp_os <- LogLogisticOs()
+
+my_link <- ttgLink()
+
+my_os <- parametrize(osmod = my_temp_os, link = my_link)
+JointModel(long = my_long_mod, os = my_os)
+```
+
+The final two steps of the package workflow are not available yet.
```
Comment (danielinteractive) [inline vignettes/Intro_to_jmpost.Rmd:72]:
```suggestion
The final two steps of the package workflow are not available yet:
```

**Review by danielinteractive — APPROVED** [review]
thanks George, looks nice! See few minor things inside. Also I did not check any formulas or the main section, intro is most important I feel... I would recommend to read again critically yourself through the whole vignette to find more things

File: vignettes/Intro_to_jmpost.Rmd [hunk 16]
```diff
@@ -59,13 +59,15 @@ The minimal usage of the `jmpost` package calls example stan code already provid
 The following chunk illustrates the absolute minimal usage of the package.
 
 ```{r, eval=FALSE}
+# The longitudinal submodel
 my_long_mod <- LongMod()
-
+# The templated survival submodel
 my_temp_os <- LogLogisticOs()
-
+# The link factor
```
Comment (danielinteractive) [inline vignettes/Intro_to_jmpost.Rmd:66]:
```suggestion
# The link between longitudinal and OS models
```
I would avoid "factor", I don't think it is a factor?

**Comment by gowerc** [comment]
Do we need the pdf to be in the repo?  Feels like we are then just tracking a load of changes unnecessarily given the source code is already being tracked. Also looks like the pdf is being manually added to the misc folder? Which means its at risk of getting out of sync.

**Pull request merged**